### PR TITLE
Documentation structure revised and content beautifulized.

### PR DESCRIPTION
**Pull request by mishaker**
## Summary
This is a work in progress of revising documentation structure and using few widgets to make the content more readable. The WIP preview is visible here https://freqtrade.readthedocs.io which currently points to my fork till the merge is accepted later.

## List of major (upcoming) changes and suggestions

1. Adding an about page which gives an introduction to Freqtrade. 
  **Suggestion:** changing "Simple High Frequency Trading Bot for crypto currencies" to "Crypto Trading Bot" as FT is not a high frequency bot. high frequency refers to trading engines usually on less than a second interval. Also "Crypto Trading Bot" is ranked better SEO wise.

1. All links should be relative.
1. Using Admonition extension for note, warnings, etc. (e.x. https://freqtrade.readthedocs.io/en/latest/pre-requisite/#6-monitor-your-docker-instance) 
1. Merging pre-requisite and installation into a same document: https://freqtrade.readthedocs.io/en/latest/pre-requisite/
1. Removing table of contents from documents as it is auto generated by MkDocs.
1. Adding Permalinks extension ( inserts an anchor at the end of each headline, which makes it possible to directly link to a subpart of the document)

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1,39 +1,67 @@
...
+
+### Hardware requirements
+To run this bot we recommend you a cloud instance with a minimum of:
+
+- 2GB RAM
+- 1GB disk space
+- 2vCPU
+
+### Software requirements
+- Python 3.6.x
```
Comment (xmatthias) [inline docs/index.md]: `python 3.6+` ... (3.7 works like a charm as well, and our docker-images are actually based on that.

File: mkdocs.yml [hunk 2]
```diff
@@ -1,13 +1,43 @@
 site_name: Freqtrade
+nav:
+    - About: index.md
```
Comment (xmatthias) [inline mkdocs.yml]: Can you rename this file to `about.md` - and put a index.md into the docs folder to point to the "real" documentation page?

That way we can redirect people from github to there without them reading through the (probably) not properly formatted docs

File: docs/backtesting.md [hunk 3]
```diff
@@ -266,5 +261,5 @@ Detailed output for all strategies one after the other will be available, so mak
 ## Next step
 
 Great, your strategy is profitable. What if the bot can give your the
-optimal parameters to use for your strategy?  
-Your next step is to learn [how to find optimal parameters with Hyperopt](https://github.com/freqtrade/freqtrade/blob/develop/docs/hyperopt.md)
+optimal parameters to use for your strategy?
+Your next step is to learn [how to find optimal parameters with Hyperopt](/hyperopt)
```
Comment (xmatthias) [inline docs/backtesting.md]: Link does not work on https://www.freqtrade.io/en/latest/backtesting/ ...

File: mkdocs.yml (hunk 2 above)
Comment (mishaker) [reply mkdocs.yml]: Didn't find any option for this (I've already tried ...). home page should be index.md for readthedocs :/

**Comment by xmatthias** [comment]
one add. thing that jumped to my mind:
we should add a link to the "main" `index.md` (the one shown on github when you go to https://github.com/freqtrade/freqtrade
also there is a "docs - passing" badge we could add to the badges list.

however I'm more than happy to do that in a followup-pr once we know this all works when switched to this repo.

**Comment by mishaker** [comment]
Yep Those will be done in a different PR:

1) adding doc badge
2) modifying readme on github
3) writing a good attractive “about” page that will be used on readme and
doc index
4) removing that “high frequency” from everywhere. I suggest we put it
“crypto trading bot”.
5) adding meta tags and headers to docs pages
6) adding pages to analytics and load data structure on it

But for this PR I guess we can stop here.
3 and 4 are items on which it would be nice if you can help me.

**Pull request merged**